### ISSUE TITLE
CI

### ISSUE THREAD

**Issue by ptheywood** [FLAMEGPU/FLAMEGPU2-model-template-cpp#15]
Basic CI to conifgure and build the template example would be a good addition, and allow users to opt-in to the same (actions shouldn't be enabled when created from a template repo by default iirc)

This would require duplication of the cuda install scripts on CI and increase maintenance burden a little.

**Comment by Robadob** [comment]
Carpentries org has a gh actions repo, with common actions which other
repos inherit. Could this be a relevant way to reduce that duplication?

**Comment by ptheywood** [comment]
Yes we could potentially do something with re-usable actions in a separate repo. It's been on my list of things to look into. 

For 3rd parties, they would have to allow 3rd party re-usable actions, or also fork the actions repo into their org/account. 

The other option is turning my very rough install cuda scripts into propper github actions, but I don't have the time nor patience to learn how to do that (it'd been requested on my dev repo for them by others previously, https://github.com/ptheywood/cuda-cmake-github-actions/issues/5). 

And that wouldn't work in our manylinux builds either which run in a non gh container)
